adsft/extraction.py

```diff
@@ -754,22 +754,67 @@ def extract_multi_content(self, translate=True, decode=True):
         return meta_out
 
 
-class PDFBoxExtractor(object):
+class PDFExtractor(object):
     def __init__(self, kwargs):
         self.ft_source = kwargs.get('ft_source', None)
         self.bibcode = kwargs.get('bibcode', None)
         self.provider = kwargs.get('provider', None)
-        self.cmd = kwargs.get('executable', proj_home + '/scripts/extract_pdf.sh') #TODO(rca) make it configurable
+        self.timeout = 120 # seconds
+        self.grobid_service = kwargs.get('grobid_service', None)
+        self.extract_pdf_script = proj_home + kwargs.get('extract_pdf_script', '/scripts/extract_pdf_with_pdftotext.sh')
 
         if not self.ft_source:
             raise Exception('Missing or non-existent source: %s', self.ft_source)
 
     def extract_multi_content(self):
-        p = Popen([self.cmd, self.ft_source], stdout=PIPE, stderr=PIPE)
+        p = Popen([self.extract_pdf_script, self.ft_source], stdout=PIPE, stderr=PIPE)
         stdout, stderr = p.communicate()
         if p.returncode != 0:
             raise Exception(stderr)
-        return {'fulltext': stdout.decode('utf8')}
+        fulltext = stdout.decode('utf8')
+        return  {
+                    'fulltext': fulltext,
+                }
+
+class GrobidPDFExtractor(object):
+    def __init__(self, kwargs):
+        self.ft_source = kwargs.get('ft_source', None)
+        self.bibcode = kwargs.get('bibcode', None)
+        self.provider = kwargs.get('provider', None)
+        self.timeout = 120 # seconds
+        self.grobid_service = kwargs.get('grobid_service', None)
+
+        if not self.ft_source:
+            raise Exception('Missing or non-existent source: %s', self.ft_source)
+
+    def extract_multi_content(self):
+        grobid_xml = ""
+        if self.grobid_service is not None:
+            try:
+                try:
+                    ft_source = open(self.ft_source, 'r')
+                except IOError, error:
+                    logger.exception("Error opening file %s: %s", self.ft_source, error)
+                logger.debug("Contacting grobid service: %s", self.grobid_service)

```

Comment:
This uses a grobid service running in the container with the public models, yes. Right now it is there just to slowly extract content to disk so that we can explore later on how well it works (not a priority task right now).